src/v2/Apps/Order/Components/AddressModal.tsx

```diff
@@ -51,34 +58,59 @@ export const AddressModal: React.FC<Props> = ({
     const errorsTrimmed = removeEmptyKeys(errors)
     return errorsTrimmed
   }
+  const { relayEnvironment } = useSystemContext()
+  const [createUpdateError, setCreateUpdateError] = useState<string>(null)
 
   return (
     <Modal title={title} show={show} onClose={closeModal}>
       <Formik
         initialValues={createMutation ? { country: "US" } : address}
         validate={validator}
-        onSubmit={values => {
+        onSubmit={(
+          values: SavedAddressType,
+          actions: FormikHelpers<SavedAddressType>
+        ) => {
+          const handleError = message => {
+            const userMessage: Record<string, string> | null =
+              SERVER_ERROR_MAP[message]
+            if (userMessage) {
+              actions.setFieldError(userMessage.field, userMessage.message)
+            } else {
+              setCreateUpdateError(GENERIC_FAIL_MESSAGE)
```

Comment:
if the error message was not found in the map, just add it to the state to be rendered on top of the form